docs/add-a-new-resource.md

```diff
@@ -0,0 +1,71 @@
+
+### Name the resource
+
+Either by creating the file manually, or using `skaff` to generate a template.
+
+All resources should be named with the following pattern: `aws_<service>_<name>`
+
+Where `<service>` is the AWS short service name that matches the key in the `serviceData` map in the `conns` package (created via the [Adding a new Service](add-a-new-service.md))
+
+Where `<name>` represents the conceptual infrastructure represented by the create, read, update, and delete methods of the service API. It should be a singular noun. For example, in an API that has methods such as `CreateThing`, `DeleteThing`, `DescribeThing`, and `ModifyThing` the name of the resource would end in `_thing`.
+
+### Fill out the Resource Schema
+
+In the `internal/service/<service>/<service>.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, choose the correct data type and supply the correct [Schema Behaviors](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors) in order to ensure Terraform knows how to correctly handle the value.
+
+Typically you will add arguments to represent the values that are under control by Terraform, and attributes in order to supply read-only values as references for Terraform. These are distinguished by Schema Behavior.
+
+Attribute names are to specified in `camel_case` as opposed to the AWS API which is `CamelCase`
+
+### Implement CRUD handlers
+These will map planned Terraform state to the AWS API call, or an AWS API response to an applied Terraform state. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
+
+### Write passing Acceptance Tests
+In order to adequately test the resource we will need to write a complete set of Acceptance Tests. You will need an AWS account for this which allows the creation of that resource. See [Writing Acceptance Tests](running-and-writing-acceptance-tests.md) for a detailed guide on how to approach these.
+
+You will need at minimum:
+
+- Basic Test - Tests full lifecycle (CRUD + Import) of a minimal configuration (all required fields, no optional).
+- Disappears Test - Tests what Terraform does if a resource it is tracking can no longer be found.
+- Per Attribute Tests - For each attribute a test should exists which tests that particular attribute in isolation alongside any required fields.
```

Comment:
This is outright incorrect, confusing "attribute" with "argument." But even correcting that, this may have been a "standard" at some point but it was never followed and is impractical or even impossible to follow. We shouldn't confuse people with it.

```suggestion
- Argument Tests - All arguments should be tested in a pragmatic way. Ensure that each argument can be initially set, updated, and cleared, as applicable. Depending on the logic and interaction of arguments, this may take one to several separate tests. 
```